compiler/compile/OMRCompilation.cpp

```diff
@@ -1285,6 +1286,29 @@ void OMR::Compilation::performOptimizations()
       _optimizer->optimize();
    }
 
+bool OMR::Compilation::incInlineDepth(TR::ResolvedMethodSymbol * method, TR::Node *callNode, bool directCall, TR_VirtualGuardSelection *guard, TR_OpaqueClassBlock *receiverClass, TR_PrexArgInfo *argInfo)
+   {
+   TR::SymbolReference *callSymRef = callNode->getSymbolReference();
+   int32_t cpIndex = callSymRef->getCPIndex();
+   TR_ByteCodeInfo &bcInfo = callNode->getByteCodeInfo();
+
+   if (self()->compileRelocatableCode())
+      {
+      TR_AOTMethodInfo *aotMethodInfo = (TR_AOTMethodInfo *)self()->trMemory()->allocateHeapMemory(sizeof(TR_AOTMethodInfo));
+      aotMethodInfo->resolvedMethod = method->getResolvedMethod();
+      aotMethodInfo->cpIndex = cpIndex;
+      aotMethodInfo->receiver = receiverClass;
+      aotMethodInfo->callSymRef = callSymRef;
+      aotMethodInfo->reloKind = self()->getReloTypeForMethodToBeInlined(guard, callNode, receiverClass);
+
+      return self()->incInlineDepth((TR_OpaqueMethodBlock*)aotMethodInfo, method, bcInfo, callSymRef, directCall, argInfo);
```

Comment:
Could we use `reinterpret_cast` here?

Reply:
Updated, see https://github.com/eclipse/omr/compare/19ad2790faa2e2c95680d830a971c583975df230..9c792c13b65a2fbabc0e9638730e8a93344dbb6f